ilc/server/app.js

```diff
@@ -23,17 +18,22 @@ const accessLogger = new AccessLogger(config);
  * @param {Registry} registryService
  */
 module.exports = (registryService, pluginManager, context) => {
-    const guardManager = new GuardManager(pluginManager);
     const reportingPlugin = reportingPluginManager.getInstance();
 
     const appConfig = Application.getConfig(reportingPlugin);
     const logger = reportingPluginManager.getLogger();
 
     const app = fastify(appConfig);
 
+    const asyncResourceSymbol = Symbol('asyncResource');
+
     app.addHook('onRequest', (req, reply, done) => {
         context.run({ request: req }, async () => {
             try {
+                const asyncResource = new AsyncResource('fastify-request-context');
```

Comment:
context.run is using Async Local Storage already maybe you could reuse it instead of running additional context ?

Reply:
There is an issue with losing context between async hooks.
See this code https://github.com/fastify/fastify-request-context/blob/master/index.js#L46